project/npda/general_functions/csv_upload.py

```diff
@@ -821,22 +821,33 @@ def save_row(row, timestamp, cohort_id):
         nhs_number = row["NHS Number"].replace(" ", "")
 
         try:
-            patient = Patient.objects.create(
-                nhs_number=nhs_number,
-                site=site,
-                date_of_birth=row["Date of Birth"],
-                postcode=row["Postcode of usual address"],
-                sex=row["Stated gender"],
-                ethnicity=row["Ethnic Category"],
-                diabetes_type=row["Diabetes Type"],
-                diagnosis_date=row["Date of Diabetes Diagnosis"],
-                death_date=(
+            patient_fields = {
+                "site": site,
+                "date_of_birth": row["Date of Birth"],
+                "postcode": row["Postcode of usual address"],
+                "sex": row["Stated gender"],
+                "ethnicity": row["Ethnic Category"],
+                "diabetes_type": row["Diabetes Type"],
+                "diagnosis_date": row["Date of Diabetes Diagnosis"],
+                "death_date": (
                     row["Death Date"] if not pd.isnull(row["Death Date"]) else None
                 ),
-                gp_practice_ods_code=row["GP Practice Code"],
-                is_valid=patient_is_valid,
-                errors=(patient_errors if patient_errors is not None else None),
-            )
+                "gp_practice_ods_code": row["GP Practice Code"],
+                "is_valid": patient_is_valid,
+                "errors": (patient_errors if patient_errors is not None else None)
+            }
+
+            # We can't use update_or_create as we need to ask AuditCohort if there is already a patient
```

Comment:
One patient can be in many AuditCohorts, and a given AuditCohort can have many patients - hence the many to many relationship. One cohort will have 4 quarters so potentially will have the same patient 4 times. A previous requirement was that we store all csv uploads so currently a given quarter can have the same patient more than once, except on upload of an existing patient to a given cohort, the previous record has the `submission_active` flag set to false and the new one set to true, so that each quarter can only have one active submission per patient. I am not sure how necessary this is, but it was something Holly insisted on in the beginning so is currently implemented.
Before I close this PR, perhaps we need a conversation about data model in general - I am keen not to overcomplicate this and perhaps you and @anchit-chandran can stop me from doing this?

Reply:
Agreed we should chat. I've written up my thoughts in the forum (RCPCH internal): https://forum.rcpch.tech/t/submission-design/154. Broadly I wonder if we should split `AuditCohort` into two: `SubmissionPeriod` and `Submission`

Reply:
Resolving as this PR no longer looks at AuditCohort - the conversation is ongoing though